docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,251 @@
+
+Pub/Sub systems have advanced topic schema, which we can take advantage of from Application Services to filter for just the Events the Application Service actual wants. Publishers of Events must add the `DeviceProfileName` and/or `DeviceName` to the topic in the form `edgex/events/<device-profile-name>/<device-name>`. This allows Application Services to filter for just the Events from the device(s) it wants by only subscribing to those `DeviceProfileNames` or the specific `DeviceNames`. Example subscribe topics if above schema is used:
+
+- **edgex/events/#**
+  - All Events 
+  - Core Data will subscribe using this topic schema
+- **edgex/events/Random-Integer-Device/#** 
+  - Only Events from devices created from the **Random-Integer-Device** device profile
+- **edgex/events/Random-Integer-Device/Random-Integer-Device1** 
+  - Only Events from the **Random-Integer-Device1** Device
+
+The MessageBus abstraction allows for multiple subscriptions, so an Application Service could specify to receive data from multiple specific device profiles or devices by creating multiple subscriptions. i.e.  `edgex/Events/Random-Integer-Device/#` and  `edgex/Events/Random-Boolean-Device/#`. Currently the App SDK only allows for a single subscription topic to be configured, but that could easily be expanded to handle a list of subscriptions. See [Configuration](#configuration) section below for details. 
+
+Core Data's existing publishing of Events would also need to be changed to use this new topic schema. One challenge with this is Core Data doesn't currently know the `DeviceProfileName` or `DeviceName` when it receives a CBOR encoded event. This is because it doesn't decode the Event until after it has published it to the MessageBus. The V2 API will be enhanced to change the AddEvent endpoint from `/event` to `/event/{profile}/{device}` so that `DeviceProfileName` and `DeviceName` are always know no matter how the request is encoded.
+
+This new topic approach will be enabled via each publisher's `PublishTopic` having the `DeviceProfileName` and `DeviceName`  added to the configured `PublishTopicPrefix`
+
+```toml
+
+PublishTopicPrefix = 'edgex/events' # /<device-profile-name>/<device-name> will be added to this Publish Topic prefix
+```
+
+See [Configuration](#configuration) section below for details. 
+
+### Configuration
+
+#### Device Services
+
+All Device services will have the following additional configuration to allow connecting and publishing to the MessageBus. As describe above in the  [MessageBus Topics](#messagebus-topics) section, the `PublishTopic` will optionally have placeholders for the `DeviceProfileName` and `DeviceName` which get replaced with the actual `DeviceProfileName` and `DeviceName`. If the place holders do not exist in the configured `PublishTopic` value, then the value is used as is.
```

Comment:
sp: As **described** above in the

Reply:
The above section states that `DeviceProfileName` and `DeviceName` **must be** provided, whereas this section suggests that they're optional. Which is it?

Reply:
No longer optional. Missed updating this section.